### DISCUSSION TITLE
Generating Typescript types with v2?

### DISCUSSION THREAD

**Discussion by zb140** [wailsapp/wails#758] [Q&A] [answered]
Hi! I just recently came across Wails, and I'm really liking it so far. Thanks for the great work!  So far I haven't had much or any trouble getting things to work, but today I decided to try the v2 alpha.  Is it possible to generate Typescript definitions for bindings in v2?  Of course I know that an alpha is expected to have missing features, but I've seen some comments and commit messages referring to it, but I couldn't figure out how to do it.

**Comment by leaanthony** [comment, marked as answer]
Hi there! There *is* a way of doing this though it's a bit hacky. If you run your app in dev mode, it will generate a node package in `frontend` wrapping your backend api. It's not currently exposed and dev mode is currently broken so we're not in an ideal position! The reason I say hacky is because it basically needs to run and it dumps the files based on reflection. This means that we lose the variable names.
A better approach, and *much* harder, is to parse the Go files of the project and generate based on that. There's a fairly robust attempt at this [here](v2/pkg/parser/generate.go) but as you can see it's complicated and I just haven't got round to making it work. I did have this as part of `wails generate` but it looks like I disabled it for some reason - I'll check why. 
In the short term, I might make `wails generate tsmodule` that will compile your project in a special mode that just dumps out the module then exits.

**Reply by zb140** [reply]
Great, thanks for the quick response! I'll keep an eye out; I'm happy to do some testing as things develop.

**Reply by leaanthony** [reply]
That would be amazing - thanks! I've got too many other things with higher priority unfortunately.
